.github/workflows/build-golang.yml

```diff
@@ -14,39 +14,51 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os-artifact: [ [ubuntu-latest, linux], [windows-latest, windows], [macos-latest, macos] ]
+        os-artifact: [ [ubuntu-latest, linux], [windows-latest, windows-gnu], [macos-latest, macos] ]
     steps:
       - uses: actions/checkout@v2
       - name: Download workflow artifact
-        uses: dawidd6/action-download-artifact@v2.14.0
+        uses: dawidd6/action-download-artifact@v2.17.0
         with:
           workflow: "build-libs.yml"
           path: ./libs/${{ matrix.os-artifact[1] }}
           github_token: ${{ secrets.GITHUB_TOKEN }}
           name: ${{ matrix.os-artifact[1] }}
-      - name: Set up Go
+      - name: Download C header artifact
+        uses: dawidd6/action-download-artifact@v2.17.0
+        with:
+          workflow: "build-libs.yml"
+          path: ./libs/${{ matrix.os-artifact[1] }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          name: C_header
+      - name: Set up Go (default)
         uses: actions/setup-go@v2
         with:
           go-version: ^1.16
       - name: Set up Python
         uses: actions/setup-python@v2
         with:
           python-version: 3.9
-      - name: Build and Test Golang
+      # Set up Okapi-specific Go runtime. Includes setting GOLANG_LD_PATH env var.
+      - name: Set up Go (Okapi)
         run: |
           go version
           go install golang.org/x/lint/golint@latest
           go install github.com/jstemmer/go-junit-report@latest
           python ../../devops/build_sdks.py
+        shell: pwsh
+        working-directory: go/okapi
+      - name: Build and Test Golang
+        run: |
           go build
           golint
-          go test -v | go-junit-report > test_output.xml
+          go test -v -exec "env DYLD_LIBRARY_PATH=${{ env.GOLANG_LD_PATH }}" | go-junit-report > test_output.xml
```

Comment:
fix 3b.  in modern Go versions, MacOS is not cool with the normal env vars, so we use `-exec.  see [issue 36572](https://github.com/golang/go/issues/36572)